BetterMD/elements/table.py

```
@@ -400,11 +504,31 @@ def from_list(cls, data:'list[str]', head:'bool'=False):
-    def prepare(self, parent = None, table=None, head:'THead|TBody|TFoot'=None, *args, **kwargs):
+    def prepare(self, parent = None, dom=None, table=None, head:'THead|TBody|TFoot'=None, *args, **kwargs):
+        print(f"TR {parent} {self.children} {self.data}")
         assert isinstance(table, Table)
+        assert isinstance(head, (THead, TBody, TFoot))
+        self.data = []
         self.table = table
-        if head: self.head = head
-        return super().prepare(parent, table=table, row=self, *args, **kwargs)
+        self.head = head
+        self.head.data.append(self)
+        ret = super().prepare(parent, dom, table=table, row=self, head=head, *args, **kwargs)
```

Ruff (0.8.2)

Star-arg unpacking after a keyword argument is strongly discouraged

(B026)